Implement #2933: Refactor Key Creation concept docs into procedures (#2989)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/record-encryption/azure-key-vault/proc-azure-key-vault-key-creation.adoc

Reviewer: ```suggestion
[source,shell]
----
az keyvault key rotation-policy update --vault-name ${VAULT_NAME} --name ${KEY_NAME} --value path/to/my_policy.json
----
+
TIP: It is recommended to use a key rotation policy.
```

= Creating Azure Key Vault keys

[role="_abstract"]
This procedure describes how to create a `key` with `wrapKey` and `unwrapKey` operations enabled for use with the Record Encryption filter. The process uses the Azure CLI to create a key and optionally apply a rotation policy.

.Prerequisites

* Access to the Azure CLI.
* A user with the _Key Vault Crypto Officer_ role (or otherwise sufficient RBAC permissions to manage keys in Azure Key Vault).

.Procedure

> . Establish a naming convention for keys within key vault for easy identification and use with the Record Encryption filter. Always use separate key vaults per application, per environment, per region; this keeps the filter’s keys separate from those used by other systems.

Reviewer: ```suggestion
. Establish a naming convention for keys within key vault for easy identification and use with the Record Encryption filter. Use separate key vaults per application, environment, and region. This ensures separation between the filter’s keys and those used by other systems.
```

> . Select a key type

Reviewer: ```suggestion
. Select a key type.
```

+
NOTE: The key types available to you in Azure Key Vault will depend on your Azure Key Vault service tier. See the _About keys_ page in the Azure Key Vault docs for more details.
+
The filter accepts the following key types and their associated wrapping algorithms:

* *Symmetric Keys (`oct` or `oct-HSM`):* Uses the quantum-resistant `A256GCM` (256-bit AES GCM) wrapping algorithm, which requires a Managed HSM subscription.
* *Asymmetric Keys (`RSA` or `RSA-HSM`):* Uses the `RSA-OAEP-256` (RSAES-OAEP with SHA-256 hash and MGF1/SHA-256 mask) wrapping algorithm, which is not quantum-resistant. This wrapping algorithm does not require a Managed HSM subscription to use (however, `RSA-HSM` is not available without a premium Azure Key Vault subscription).

> . Create a key with the Azure CLI

Reviewer: ```suggestion
. Create a key with the Azure CLI:
```
